aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientTest.java

```diff
@@ -313,7 +313,7 @@ public void testSignUp() throws Exception {
 
```

Comment:
```
try {
    UUID.fromString(id);
} catch (IllegalArgumentException error) {
    fail("userSub must be a valid UUID");
}
```

Reply:
oh this was in response to an unrelated section of the code:
> assertNotNull(userSub);
// Validate that the userSub is a valid UUID
        assertEquals(36, userSub.length());
        assertEquals(5, userSub.split("-").length);
        assertEquals(8,  userSub.split("-")[0].length());
        assertEquals(4,  userSub.split("-")[1].length());
        assertEquals(4,  userSub.split("-")[2].length());
        assertEquals(4,  userSub.split("-")[3].length());
        assertEquals(12,  userSub.split("-")[4].length());

Reply:
EDIT: nevermind! This suggestion is not a good solution according to this [stack overflow](https://stackoverflow.com/questions/20041051/how-to-judge-a-string-is-uuid-type) discussion! Another way to test would be 
```
assertEquals(userSub, UUID.fromString(userSub).toString());
```

Reply:
Got it - thanks for the suggestion but I think we should leave existing code as is unless it's broken. :)